train.py

```diff
@@ -107,10 +110,38 @@ def main(args):
         # save checkpoint
         if epoch_itr.epoch % args.save_interval == 0:
             save_checkpoint(args, trainer, epoch_itr, valid_losses[0])
+
+        epoch_itr = reload_train(args, epoch_itr, max_positions, task)
     train_meter.stop()
     print('| done training in {:.1f} seconds'.format(train_meter.sum))
 
 
+def reload_train(args, epoch_itr, max_positions, task):
+    print("| Reloading train data with epoch: ", epoch_itr.epoch)
+    timer = StopwatchMeter()
+    timer.start()
+    load_dataset_splits(args, task, [args.train_subset], epoch=epoch_itr.epoch)
+    epoch_state_dict = epoch_itr.state_dict()
+    epoch_itr = task.get_batch_iterator(
+        dataset=task.dataset(args.train_subset),
+        max_tokens=args.max_tokens,
+        max_sentences=args.max_sentences,
+        max_positions=max_positions,
+        ignore_invalid_inputs=True,
+        required_batch_size_multiple=args.required_batch_size_multiple,
+        seed=args.seed,
+        num_shards=args.distributed_world_size,
+        shard_id=args.distributed_rank,
+        num_workers=args.num_workers,
+    )
+    # don't fast-forward, if the data is sharded
+    if len(args.data.split(":")) > 0:
```

Comment:
is this supposed to be >1? otherwise when does it == 0?

Separately, we should exit this function early when len(...) == 1.